libres/lib/res_util/block_fs.cpp

```diff
@@ -634,53 +503,25 @@ block_fs_type *block_fs_mount(const fs::path &mount_file, int block_size,
     return block_fs;
 }
 
-static file_node_type *block_fs_get_new_node(block_fs_type *block_fs,
-                                             const char *filename,
-                                             size_t min_size) {
-
-    long int offset;
-    int node_size;
-    file_node_type *new_node;
-
-    {
-        div_t d = div(min_size, block_fs->block_size);
-        node_size = d.quot * block_fs->block_size;
-        if (d.rem)
-            node_size += block_fs->block_size;
-    }
-
+static BlockMeta block_fs_get_new_node(block_fs_type *block_fs,
```

Comment:
This fseeks to the end of the file and gives you the block, so name is confusing and there is two things happening.